### PR TITLE
Job, task, taskflow log notifications

### PR DESCRIPTION
**Pull request by TristanWright**
requires https://github.com/Kitware/cumulus/pull/231

In the initial commit only taskflow logs are being streamed

- [x] Taskflow stream
- [x] Job stream
- [x] Task stream
- [x] Cluster stream, remove subscription model

---
Closes:
- #401 task log doesn't update on error
- #393 page refresh required for ec2 log to appear 
- #332 job or task logs closing after log update
- #281 jobs with old state after transition
- #430 log colors
- #431 exlcude /test folder

### CODE REVIEW

File: src/utils/logSort.js [hunk 1]
```diff
@@ -0,0 +1,3 @@
+export default function logSort(entryA, entryB) {
+  return Date.parse(entryA.created) < Date.parse(entryB.created);
```
Comment (TristanWright) [inline src/utils/logSort.js]: When I was testing this I was getting some events out of order so we're sorting each time here. Something more clever should probably be done. e.g. 
```py
entries = [{created: 2:30}, {created: 4:15}]
receive new logEntry a {created: 3:00}, 
endEntry = []
iterate backwards through entries, get entry b
  endEntry.unshift(b)
  if a.created < b.created: endEntry.unshift(a), break
# endEntry = [a, {created: 4:15}]
entries.splice(0, entries.length - endEntry.length) # entries  = [{created: 2:30}]
entries = entries.concat(endEntry) #entrie = [{created: 2:30}, a, {created: 4:15}]
```
Which is probably faster than sorting them every time especially as the logs get longer?

**Comment by TristanWright** [comment]
Cluster logs stream well, however when the log is open it can have some odd behavior. Certainly still a WIP but the core is there.

**Comment by TristanWright** [comment]
In reference to the log sorting before. Here are some out of order logs:

screen shot 2016-06-22 at 17 57 24

**Comment by TristanWright** [comment]
Odd behavior is that the log will simply go empty sometimes. Closing and opening the panel fixes it, but why does it go blank in the first place?

**Comment by TristanWright** [comment]
This branch is ready for the most part and I'm tempted to take off the WIP tag. I'm looking at two small things within ExecutionUnit:

- [ ] logs appearing out of order
- [ ] autoscroll not working

**Comment by cjh1** [comment]
@TristanWright It seems that the taskflow tasks are not showing up on this branch. This is what I see when that taskflow is complete:

task

After a page refresh I see the tasks:

task_refresh

**Comment by TristanWright** [comment]
I was seeing this yesterday but I thought I had cleared it up. I'll look at
it again.

On Tue, Jun 28, 2016 at 1:08 PM, Chris Harris <notifications@github.com>
wrote:

> @TristanWright <https://github.com/TristanWright> It seems that the
> taskflow tasks are not showing up on this branch. This is what I see when
> that taskflow is complete:
>
> [image: task]
> <https://cloud.githubusercontent.com/assets/1031053/16428901/100826e6-3d42-11e6-8938-27caaa3365ae.png>
>
> After a page refresh I see the tasks:
>
> [image: task_refresh]
> <https://cloud.githubusercontent.com/assets/1031053/16428907/170dca68-3d42-11e6-9057-2342014fef78.png>
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/Kitware/HPCCloud/pull/409#issuecomment-229150844>, or mute
> the thread
> <https://github.com/notifications/unsubscribe/AMs36S0Zr1QWfcMnJcv6o7vEteIYngP1ks5qQXEhgaJpZM4ImBV0>
> .
>



-- 
- Tristan Wright

*R&D Engineer*
*Kitware Inc.*

**Comment by TristanWright** [comment]
When I added rate limiting calling `taskflows/{taskflowid}/tasks`, I was doing it too strictly. API calls now limited by taskflowId.

**Comment by cjh1** [comment]
LGTM

**Pull request merged**